os_dep.c

```diff
@@ -3479,8 +3479,9 @@ GC_write_fault_handler(struct _EXCEPTION_POINTERS *exc_info)
     }
 
 #    ifdef CHERI_PURECAP
-    hdr *hhdr = HDR(h); 
-    UNPROTECT(hhdr->hb_block, GC_page_size);
+    hdr *hhdr = HDR(GC_prev_block(h));
```

Comment:
I think this change is actually masking the issue -- from my debugging the signal handler is actually not returning the address that triggered the fault handler in the first place.